### ISSUE TITLE
Cannot run integration test while having the vim extension installed 

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#27537]
Testing #27456

I have the vim extension installed and disabled (always), but it looks like the integration test picks it up and enables it. This causes the integration test to fail in the Data Migration -> checks if the Untitled file is restored migrating from stable to latest test

I will continue by uninstalling the vim extension.

**Comment by roblourens** [comment]
Removing 'bug', I don't think a test failure needs verification.

**Comment by michelkaporin** [comment]
@roblourens This test failure was caused due to missing implementation in smoke test to tackle the problem @alexandrudima faced, not because it failed due to a bug in VS Code. It would be still great to verify it.

**Comment by chrmarti** [comment]
I just ran the tests from master (d167d92) with Vim installed, but always disabled:

```
  38 passing (31m)
  4 failing



  1) Smoke Test Suite Data Loss verifies that contents of the dirty files are restored after 'hot exit':

      AssertionError: '[' == 'Hello, Code'
      + expected - actual

      -[
      +Hello, Code

      at Context.<anonymous> (out/tests/data-loss.js:71:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/data-loss.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)


  2) Smoke Test Suite Configuration and views turns off editor line numbers and verifies the live change:

      AssertionError: 4 == 0
      + expected - actual

      -4
      +0

      at Context.<anonymous> (out/tests/configuration-views.js:46:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/configuration-views.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)


  3) Smoke Test Suite Git stages, commits change to 'app.js' locally and verifies outgoing change:
     element not visible
  Error


  4) Smoke Test Suite Localization starts with 'DE' locale and verifies title and viewlets text is in German:

      AssertionError: 'quellcodeverwaltung' == 'quellcodeverwaltung: git'
      + expected - actual

      -quellcodeverwaltung
      +quellcodeverwaltung: git

      at Context.<anonymous> (out/tests/localization.js:42:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/localization.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)
```

![27537](https://user-images.githubusercontent.com/9205389/26846909-10810ff8-4ab0-11e7-96a2-0b13cc3339e7.png)

Reopening for determining if this is still the same issue.

**Comment by Tyriar** [comment]
I tested this as well, couldn't get it working.

**Comment by chrmarti** [comment]
I should add: I'm on Mac with several extensions enabled besides the disabled Vim one:
- VSCode Version: Code - Insiders 1.13.0-insider (eaa887c507e1614f90c9f04a30ceda775759e814, 2017-06-06T10:23:09.381Z)
- OS Version: Darwin x64 15.6.0

**Comment by michelkaporin** [comment]
Yes, in fact the notification message from Vim prevents the smoke test performing actions on some of the elements which clarifies 1, 2 and 3rd error. My fix was only for 'Data Migration' area of the test, and it seems that Vim decided to show this notification since the time when @alexandrudima performed the test. The proper full fix for it will be addressed in #27921.

However I could not see why the last test failed for you @chrmarti. I've rerun the smoke test also on OS X and it did not fail. Which executable did you run over? Do you have any other extensions installed?

When you start executable yourself and go to SCM viewlet, do you see the following heading or different one?
![image](https://user-images.githubusercontent.com/2239563/26872132-c610e102-4b75-11e7-9d53-f068a59e8e15.png)

**Comment by chrmarti** [comment]
I see that heading, not sure what could have caused this. I do have other extensions installed. Should the test be run without any extensions installed? If so, maybe the test could just run with all extensions disabled?

**Comment by michelkaporin** [comment]
@chrmarti that is exactly what I plan to do in #27921. I am moving this issue to June milestone to verify again, once #27921 is implemented.

**Comment by michelkaporin** [comment]
It should be working now and ready for verification 👍

**Comment by chrmarti** [comment]
Reran with the latest tests from master, 1.12.2 as stable and:
- VSCode Version: Code - Insiders 1.13.0-insider (c8123b036b66ca6e319bae00bfd7c34bb1ad977b, 2017-06-06T21:23:06.005Z)
- OS Version: Darwin x64 15.6.0
- Extensions: 

With the following failures:

```
  1) Smoke Test Suite Search "before each" hook for "searches only for *.js files & checks for correct result number":
     Error: Timeout of 360000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)


  2) Smoke Test Suite Git verifies current changes are picked up by Git viewlet:
     AssertionError: '1' == 2
      at Context.<anonymous> (out/tests/git.js:40:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/git.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)


  3) Smoke Test Suite Git verifies 'app.js' diff viewer changes:
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)



  4) Smoke Test Suite Git stages 'app.js' changes and checks stage count:
     An element could not be located on the page using the given search parameters.
  Error


  5) Smoke Test Suite Git stages, commits change to 'app.js' locally and verifies outgoing change:
     An element could not be located on the page using the given search parameters.
```

I wasn't at my desk when the Search one failed. The Git ones seem to be caused by app.js not having any changes. Screenshot from when these failed:
![image](https://user-images.githubusercontent.com/9205389/26898437-c9ddb7f0-4b80-11e7-829b-b20892fdd96e.png)

**Comment by michelkaporin** [comment]
@chrmarti I cannot reproduce it on my side running the same configuration as yours. 

It seems that `app.js` of the vscode-smoketest-express folder is not saved in `Search replaces first search result with a replace term` test. That's why you are hitting Git test failures. From your output I can see that something wrong happened during `Search searches only for *.js files & checks for correct result number` test that lead to this issue.

Would you be able to capture your screen from CSS until Git tests included please?
To speed the process up just comment out lines 22-25 and 31-35 in https://github.com/Microsoft/vscode/blame/master/test/smoke/src/test.ts#L28,
including their respective imports.

**Comment by chrmarti** [comment]
Using the latest from master (had to change the Electron types dependency to `~1.4.37`, so it doesn't upgrade to 1.6).

The Search failures did not occur (I also tried running from the beginning.), the Git ones show other errors:
```
1) Smoke Test Suite Git verifies current changes are picked up by Git viewlet:
     TypeError: Cannot read property 'key' of undefined
      at SpectronApplication.command (out/spectron/application.js:149:29)
      at Git.openGitViewlet (out/areas/git.js:23:30)
      at Context.<anonymous> (out/tests/git.js:41:27)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/git.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:135:7)


  2) Smoke Test Suite Git verifies 'app.js' diff viewer changes:
     TypeError: Cannot read property 'key' of undefined
      at SpectronApplication.command (out/spectron/application.js:149:29)
      at Git.openGitViewlet (out/areas/git.js:23:30)
      at Context.<anonymous> (out/tests/git.js:48:27)
      at Generator.next (<anonymous>)
      at out/tests/git.js:11:71
      at __awaiter (out/tests/git.js:7:12)
      at Context.<anonymous> (out/tests/git.js:47:20)


  3) Smoke Test Suite Git stages 'app.js' changes and checks stage count:
     TypeError: Cannot read property 'key' of undefined
      at SpectronApplication.command (out/spectron/application.js:149:29)
      at Git.openGitViewlet (out/areas/git.js:23:30)
      at Context.<anonymous> (out/tests/git.js:58:27)
      at Generator.next (<anonymous>)
      at out/tests/git.js:11:71
      at __awaiter (out/tests/git.js:7:12)
      at Context.<anonymous> (out/tests/git.js:57:20)


  4) Smoke Test Suite Git stages, commits change to 'app.js' locally and verifies outgoing change:
     TypeError: Cannot read property 'key' of undefined
      at SpectronApplication.command (out/spectron/application.js:149:29)
      at Git.openGitViewlet (out/areas/git.js:23:30)
      at Context.<anonymous> (out/tests/git.js:69:27)
      at Generator.next (<anonymous>)
      at out/tests/git.js:11:71
      at __awaiter (out/tests/git.js:7:12)
      at Context.<anonymous> (out/tests/git.js:68:20)
```

Will let you know where I put the recording.

**Comment by michelkaporin** [comment]
@chrmarti very strange that the search failure suddenly has gone, I can explain it only if you intervened in the test run. 

Thanks for posting the latest log, apparently we had changed our command for opening an SCM viewlet (from `workbench.view.git` to `workbench.view.scm`) recently, so that I had to take it into account. Corrected it in 631e0a74eece2f5b7829c2723b5249683a5af00d.

Just to clarify - I am using key bindings based on commands we specify in https://github.com/Microsoft/vscode-docs/tree/master/scripts/keybindings, and these were updated recently to match 1.13 release, thus you got these errors.
